### PR TITLE
Add PATCH endpoint for virtual services

### PR DESCRIPTION
**Pull request by jsravn**
Requires the same parameters as PUT, and updates the existing service in
place. If the service doesn't exist, it will create it. This allows
automation to just send a PATCH always.

### CODE REVIEW

**Comment by jsravn** [comment]
I'm testing it out locally, there are some issues. I'll push the fixes soonish.

File: core/context.go [hunk 1]
```diff
@@ -216,11 +216,24 @@ func (ctx *Context) updateService(vsID string, opts *ServiceOptions) error {
 		return err
 	}
 
-	if _, exists := ctx.services[vsID]; !exists {
+	var old *service
```
Comment (alangibson01) [inline core/context.go]: Do we need this line?

Comment (jsravn) [reply core/context.go]: Nope I'll remove it.

**Comment by jsravn** [comment]
@alangibson01 Can you be more specific what I need to update?

**Pull request merged**